test/integration/vpc_e2e_test.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"path"
+	"reflect"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/wait"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"antrea.io/nephe/apis/crd/v1alpha1"
+	cloudUtils "antrea.io/nephe/pkg/cloud-provider/utils"
+	"antrea.io/nephe/test/utils"
+)
+
+var _ = Describe(fmt.Sprintf("%s,%s: Cloud Resource VPC", focusAws, focusAzure), func() {
+
+	var (
+		ns = staticVMNS
+	)
+
+	BeforeEach(func() {
+		logf.Log.Info("BeforeEach", "namespace", ns)
+	})
+
+	AfterEach(func() {
+		result := CurrentGinkgoTestDescription()
+		if result.Failed {
+			if len(supportBundleDir) > 0 {
+				logf.Log.Info("Collect support bundles for test failure.")
+				fileName := utils.GenerateNameFromText(result.FullTestText, testFocus)
+				utils.CollectSupportBundle(kubeCtl, path.Join(supportBundleDir, fileName), cloudVPC, withAgent)
+			}
+			if preserveSetupOnFail {
+				logf.Log.V(1).Info("Preserve setup on failure")
+				return
+			}
+		}
+
+		// cleanup the test configuration by deleting the namespace.
+		if ns == nil {
+			return
+		}
+		logf.Log.V(1).Info("Delete namespace", "ns", ns.Name)
+		err := k8sClient.Delete(context.TODO(), ns)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("Basic Test for VPC", func() {
+		vmIDs := cloudVPC.GetVMs()
+		vpcID := cloudVPC.GetVPCID()
+		vpcName := cloudVPC.GetVPCName()
+		cloudProvider := strings.Split(cloudProviders, ",")[0]
+		kind := reflect.TypeOf(v1alpha1.VirtualMachine{}).Name()
+		// Create a Namespace
+		if len(ns.Status.Phase) == 0 {
+			// Create ns if not already present.
+			ns.Name = "test-cloud-e2e-" + fmt.Sprintf("%x", rand.Int())
+			ns.Labels = map[string]string{"name": ns.Name}
+			err := k8sClient.Create(context.TODO(), ns)
+			Expect(err).ToNot(HaveOccurred())
+			logf.Log.V(1).Info("Created namespace", "ns", ns.Name)
+		}
+
+		// Configure CPA.
+		logf.Log.V(1).Info("Configure and create CPA")
```

Comment:
mark the step with By () and validation with logger

Reply:
Added

Reply:
line 120 is repeated log of 119